osbrain/helper.py

```diff
@@ -192,3 +192,36 @@ def wait_agent_attr(agent, name='received', length=None, data=None, value=None,
             break
         time.sleep(0.01)
     return False
+
+
+def synchronize_sync_pub(server, server_alias, client, client_alias):
+    '''
+    Create a SYNC_PUB/SYNC_SUB channel and connect both agents.
+
+    Make sure they have stablished the PUB/SUB communication within the
+    SYNC_PUB/SYNC_SUB channel before returning. This will guarantee that
+    no PUB messages are lost.
+    '''
+    def assert_receive(agent, message, topic=None):
+        try:
+            agent.get_attr('_tmp_attr')
+            agent.set_attr(_tmp_attr=True)
+        except AttributeError:  # Attribute already deleted
+            pass
+
+    uuid = client.get_uuid_used_as_alias_for_sub_in_sync_pub(client_alias)
+
+    # Set a temporary custom handler
+    client.set_attr(_tmp_attr=False)
+    original_handler = client.get_handler(uuid)
+    client.subscribe(uuid, assert_receive)
+
+    # Send messages through the PUB socket until the client receives them
+    server.each(0.1, 'send', server_alias, 'Synchronize', alias='_tmp_timer')
+    assert wait_agent_attr(client, name='_tmp_attr', value=True, timeout=5)
+    server.stop_timer('_tmp_timer')
+
+    # Restore the original handler, now that the connection is guaranteed
+    client.subscribe(uuid, original_handler)
+
+    client.del_attr('_tmp_attr')
```

Comment:
I think it would not be possible to synchronize them through requests. I am thinking about the case in which a request might change the internal state of the SYNC_PUB agent, and for those cases, an extra hack around would have to be made, I think (changing the handler of the SYNC_PUB momentarily for the requests).

Perhaps what we could do is explain that some messages might be lost in the process or that it should only be used for testing purposes. Now I'm thinking that perhaps the function does not even belong there. In the end, I came up with it to make some tests more consistent.

Any thoughts?